server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/model/assertion/NEBAssertionTests.java

```diff
@@ -0,0 +1,1479 @@
+      Map<Long,Integer> cvrDiscrepancies, Integer oneVoteOver, Integer oneVoteUnder,
+      Integer twoVoteOver, Integer twoVoteUnder, Integer other)
+  {
+    log(LOGGER, String.format("testNEBEstimatedVaryingSamples[%d;%f;%f;%d;%d:%d;%d;%d]",
+        auditedSamples, riskLimit, dilutedMargin, oneVoteOver, oneVoteUnder, twoVoteOver,
+        twoVoteUnder, other));
+
+    Assertion a = createNEBAssertion("W", "L", TC, rawMargin, dilutedMargin.doubleValue(),
+        difficulty.doubleValue(), cvrDiscrepancies, oneVoteOver, oneVoteUnder, twoVoteOver,
+        twoVoteUnder, other);
+
+    // Note that the way optimistic/estimated sample computation is performed is that
+    // there is an assumption that the optimistic calculation has occurred prior to
+    // 'computeEstimatedSamplesToAudit' being called. This fits with the existing design
+    // of how plurality sample size computation is done.
+    a.computeOptimisticSamplesToAudit(riskLimit);
+    final int result = a.computeEstimatedSamplesToAudit(auditedSamples);
+    final int expected = AssertionTests.estimated(riskLimit, dilutedMargin.doubleValue(),
+        oneVoteOver, twoVoteOver, oneVoteUnder, twoVoteUnder, Audit.GAMMA, auditedSamples);
+
+    assertEquals(result, expected);
+    assertEquals(a.estimatedSamplesToAudit.intValue(), expected);
+  }
+
+  /**
+   * Test Assertion::recordDiscrepancy(CVRAuditInfo) in the context where no discrepancy has
+   * been computed for the given CVR-ACVR pair, and the assertion has no prior recorded
+   * discrepancies. It should not change the assertion's discrepancy counts. What
+   * recordDiscrepancy() does is look for the CVRAuditInfo's ID in its cvrDiscrepancy map. If it is
+   * there, the value matching the ID key is retrieved and the associated discrepancy type
+   * incremented. If it is not there, then a runtime exception is thrown.
+   */
+  @Test(expectedExceptions = {RuntimeException.class})
+  public void testNEBRecordNoMatch1(){
+    log(LOGGER, "testNEBRecordNoMatch1");
+    CVRAuditInfo info = new CVRAuditInfo();
+    info.setID(1L);
+
+    Assertion a = createNEBAssertion("W", "L", TC, 50, 0.1,
+        8, Map.of(), 0, 0, 0, 0, 0);
+
+    a.recordDiscrepancy(info);
+  }
+
+  /**
+   * Test Assertion::recordDiscrepancy(CVRAuditInfo) in the context where no discrepancy has
+   * been computed for the given CVR-ACVR pair, and the assertion has some discrepancies recorded
+   * already. A runtime exception should be thrown.
+   */
+  @Test(expectedExceptions = {RuntimeException.class})
+  public void testNEBRecordNoMatch2(){
+    log(LOGGER, "testNEBRecordNoMatch2");
+    CVRAuditInfo info = new CVRAuditInfo();
+    info.setID(1L);
+
+    Assertion a = createNEBAssertion("W", "L", TC, 50, 0.1,
+        8, Map.of(2L, -1, 3L, 1), 1, 1,
+        0, 0, 0);
+
+    a.recordDiscrepancy(info);
+  }
+
+  /**
+   * Test Assertion::recordDiscrepancy(CVRAuditInfo) for a one vote overvote.
```

Comment:
I think "overvote" should be replaced with "overstatement" everywhere. (Unless you meant 'overvote' and I'm confused.)

Reply:
(I think the corla code uses "overcount" and "overstatement" interchangeably, but "overvote" is a different thing, meaning that the voter chose too many options.)

Reply:
Same comment for NEBAssertionTests.